megatron/training/utils.py

```diff
@@ -228,6 +229,210 @@ def get_ltor_masks_and_position_ids(data,
 
     return attention_mask, loss_mask, position_ids
 
+def get_batch_on_this_tp_rank_original(data_iterator):
```

Comment:
What's the difference between this function and the `get_batch_on_this_tp_rank` function in this file?

Reply:
fixed